pkg/util/containers/metrics/provider/provider.go

```diff
@@ -65,9 +65,41 @@ var (
 	}
 )
 
+// RuntimeFlavor is a typed string for supported container runtime flavors
+type RuntimeFlavor string
+
+// Known container runtime flavors
+const (
+	RuntimeFlavorDefault RuntimeFlavor = ""
+	RuntimeFlavorKata    RuntimeFlavor = "kata"
+)
+
+// RuntimeMetadata contains the runtime flavor and runtime name
+type RuntimeMetadata struct {
+	flavor  RuntimeFlavor
+	runtime Runtime
+}
+
+// NewRuntimeMetadata returns a new RuntimeMetadata
+func NewRuntimeMetadata(runtime, flavor string) RuntimeMetadata {
+	return RuntimeMetadata{
+		flavor:  RuntimeFlavor(flavor),
+		runtime: Runtime(runtime),
+	}
+}
+
+// String returns the runtime compose.
+func (r *RuntimeMetadata) String() Runtime {
+	if r.flavor == RuntimeFlavorKata {
+		return r.runtime + Runtime("-") + Runtime(RuntimeFlavorKata)
```

Comment:
I don't think you need to be specific. If `r.flavor != nil` you concatenate, if not you return `r.runtime`.